src/ctest.ts

```diff
@@ -432,70 +422,70 @@ export class CTestDriver implements vscode.Disposable {
                 }
                 return returnCode;
             } else {
-                runCTestImplArgs.push([_driver, _ctestPath, _ctestArgs, test, customizedTask, consumer]);
-                driverSet.add(_driver);
-                ctestPathSet.add(_ctestPath);
-                ctestArgsSet.add(_ctestArgs);
-                customizedTaskSet.add(customizedTask);
-                consumerSet.add(consumer);
+                if (!driverMap.has(_driver.sourceDir)) {
+                    driverMap.set(_driver.sourceDir, { driver: _driver, ctestPath: _ctestPath, ctestArgs: _ctestArgs, tests: [test] });
+                } else {
+                    const d = driverMap.get(_driver.sourceDir);
+                    driverMap.set(_driver.sourceDir, { driver: d!.driver, ctestPath: d!.ctestPath, ctestArgs: d!.ctestArgs, tests: d!.tests.concat(test) });
```

Comment:
Do we assume here that all tests of the same `_driver.sourceDir` automatically share the same `driver`, `ctestPath`, and `ctestArgs`?